src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js

```diff
@@ -1,122 +1,25 @@
-            // the store in order for the `Form` to function. For these reasons we generate it once here.
-            formId: `searchPageForm-${id}-${index}`,
-            searchScope,
-            minAttributesRequiredToSearch,
-        })),
-});
-
-export const searchScopes = {
-    PROGRAM: 'PROGRAM',
-    TRACKED_ENTITY_TYPE: 'TRACKED_ENTITY_TYPE',
-};
+import type { OwnProps, Props, PropsFromRedux } from './SearchPage.types';
 
 const mapStateToProps = (state: ReduxState): PropsFromRedux => {
-    const { currentSelections, activePage, searchPage: { searchStatus, generalPurposeErrorMessage } } = state;
-
-    const trackedEntityTypesWithCorrelatedPrograms =
-      [...programCollection.values()]
-          .filter(program => program instanceof TrackerProgram)
-          // $FlowFixMe
-          .reduce((acc, {
-              id: programId,
-              name: programName,
-              trackedEntityType: {
-                  id: trackedEntityTypeId,
-                  name: trackedEntityTypeName,
-                  searchGroups: trackedEntityTypeSearchGroups,
-              },
-              searchGroups,
-          }: TrackerProgram) => {
-              const accumulatedProgramsOfTrackedEntityType =
-                acc[trackedEntityTypeId] ? acc[trackedEntityTypeId].programs : [];
-              return {
-                  ...acc,
-                  [trackedEntityTypeId]: {
-                      trackedEntityTypeId,
-                      trackedEntityTypeName,
-                      trackedEntityTypeSearchGroups,
-                      programs: [
-                          ...accumulatedProgramsOfTrackedEntityType,
-                          { programId, programName, searchGroups },
-                      ],
-
-                  },
-              };
-          }, {});
-
-    const preselectedProgram = Object.values(trackedEntityTypesWithCorrelatedPrograms)
-        // $FlowFixMe https://github.com/facebook/flow/issues/2221
-        .map(({ programs }) =>
-            programs.find(({ programId }) => programId === currentSelections.programId))
-        .filter(program => program)[0];
-
-    const availableSearchOptions = Object.values(trackedEntityTypesWithCorrelatedPrograms)
-        // $FlowFixMe https://github.com/facebook/flow/issues/2221
-        .reduce((acc, { trackedEntityTypeId, trackedEntityTypeName, trackedEntityTypeSearchGroups, programs }) => ({
-            ...acc,
-            [trackedEntityTypeId]: buildSearchOption(trackedEntityTypeId, trackedEntityTypeName, trackedEntityTypeSearchGroups, searchScopes.TRACKED_ENTITY_TYPE),
-            ...programs.reduce((accumulated, { programId, programName, searchGroups }) => ({
-                ...accumulated,
-                [programId]: buildSearchOption(programId, programName, searchGroups, searchScopes.PROGRAM),
-            }), {}),
-        }), {});
+    const { activePage } = state;
 
     return {
```

Comment:
basically this could also be removed. I simply cannot imagine how at the moment. 🤔 maybe you have any ideas?